### ISSUE TITLE
How to predict with onnx model

### ISSUE THREAD

**Issue by yuanshuai666** [ultralytics/ultralytics#19858]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Dear Author,
Could you please advise how to perform inference on an ONNX model using Ultralytics YOLOv8?
Could you also provide a code example?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yuanshuai666, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training or ❓ Question, please provide as much context as possible, including your environment details (e.g., Python and PyTorch versions), and any relevant code snippets to assist in resolving your inquiry.

For ONNX-specific guidance, you can also explore our [Export](https://docs.ultralytics.com/modes/export/) and [Inference](https://docs.ultralytics.com/modes/predict/) documentation sections for useful details. If additional clarification is needed, an Ultralytics engineer will assist you soon.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can load the ONNX model normally just like `.pt` model in Ultralytics 

```
from ultralytics import YOLO 
model = YOLO("yolo11n.onnx")
results= model("image.jpg")
```

**Comment by yuanshuai666** [comment]
Dear Author,
Thanks for your reply!
How can we avoid accuracy loss when converting a .pt model to an engine model?
Some objects can be detected with the .pt model, but after converting to the engine model, they can no longer be recognized.

**Comment by Y-T-G** [comment]
You can export with `dynamic=True`

**Comment by yuanshuai666** [comment]
> You can export with `dynamic=True`

Thank you for your reply. As far as I know, the dynamic=True parameter controls whether the model supports dynamic input sizes. However, both the training and testing image sizes for my model are fixed at 1472×1472. Will this have any impact on improving the accuracy after converting the model to an engine?

**Comment by Y-T-G** [comment]
Where are you getting accuracy difference? Are you using Ultralytics for inference?

**Comment by yuanshuai666** [comment]
> Where are you getting accuracy difference? Are you using Ultralytics for inference?

Yes, I use Ultralytics for inference. Below is my inference code:
from ultralytics import YOLO
from PIL import Image
import cv2



model = YOLO("./runs/detect/train10/weights/best.engine", task="detect")
results = model.predict("./1111/20230201-10-59-40_699.jpg", imgsz=1472)

print(results)
# Visualize the results
for idx, result in enumerate(results):
    annotated_image = result.plot()
    
 
    window_name = f"Detection Result {idx}"
    cv2.imshow(window_name, annotated_image)
    cv2.waitKey(0)  
    cv2.destroyWindow(window_name)

cv2.destroyAllWindows()
----------------------
from PIL import Image
import cv2
from ultralytics import YOLO

# Load a pretrained YOLO11n model
model = YOLO("./runs/detect/train10/weights/best.pt")
# print(model.model.args)
# Run inference on 'bus.jpg'
results = model(["./1111/20230201-10-59-40_699.jpg"], imgsz=1472)  # results list
# print(results)
# Visualize the results
for idx, result in enumerate(results):
    # 使用 result.plot() 获取带检测框的图像
    annotated_image = result.plot()
    
    # 显示图像窗口
    window_name = f"Detection Result {idx}"
    cv2.imshow(window_name, annotated_image)
    cv2.waitKey(0)  # 等待按键后继续
    cv2.destroyWindow(window_name)

cv2.destroyAllWindows()

**Comment by yuanshuai666** [comment]
> Where are you getting accuracy difference? Are you using Ultralytics for inference?

Yes, I use Ultralytics for inference. Below is my inference code:
`from PIL import Image
import cv2
from ultralytics import YOLO

# Load a pretrained YOLO11n model
model = YOLO("./runs/detect/train10/weights/best.pt")
# print(model.model.args)
# Run inference on 'bus.jpg'
results = model(["./1111/20230201-10-59-40_699.jpg"], imgsz=1472)  # results list
# print(results)
# Visualize the results
for idx, result in enumerate(results):
    # 使用 result.plot() 获取带检测框的图像
    annotated_image = result.plot()
    
    # 显示图像窗口
    window_name = f"Detection Result {idx}"
    cv2.imshow(window_name, annotated_image)
    cv2.waitKey(0)  # 等待按键后继续
    cv2.destroyWindow(window_name)

cv2.destroyAllWindows()`

`from ultralytics import YOLO
from PIL import Image
import cv2



model = YOLO("./runs/detect/train10/weights/best.engine", task="detect")
results = model.predict("./1111/20230201-10-59-40_699.jpg", imgsz=1472)

print(results)
# Visualize the results
for idx, result in enumerate(results):
    # 使用 result.plot() 获取带检测框的图像
    annotated_image = result.plot()
    
    # 显示图像窗口
    window_name = f"Detection Result {idx}"
    cv2.imshow(window_name, annotated_image)
    cv2.waitKey(0)  # 等待按键后继续
    cv2.destroyWindow(window_name)

cv2.destroyAllWindows()`

**Comment by Y-T-G** [comment]
If you export with dynamic, it will be the same
